File: cpp/card.cpp
#include "card.h"

Card::Card(int r, char s) : rank(r), suit(s) {}

std::string Card::toString() const {
	std::string rankStr;
	if (rank == 14) {
		 rankStr = "A";
	} else if (rank == 13) {
		 rankStr = "K";
	} else if (rank == 12) {
		rankStr = "Q";
	} else if (rank == 11) {
		rankStr = "J";
	} else {
		rankStr = std::to_string(rank);
	}
	return rankStr + suit;
}

File: julia/Card.jl
# Card.jl
module CardModule

export Card, to_string

# Card struct
struct Card
	rank::Int
    	suit::Char
end

# Convert the Card object to string representation
function to_string(card::Card)
	rank_str = if card.rank == 14
		"A"
	elseif card.rank == 13
        	"K"
    	elseif card.rank == 12
        	"Q"
    	elseif card.rank == 11
        	"J"
    	else
        	string(card.rank)
    	end
    	return rank_str * string(card.suit)
end

end # module




